test/components/legendTests.ts

```diff
@@ -350,3 +350,183 @@ describe("Legend", () => {
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "horizontal");
+    legend.renderTo(svg);
+
+    assertBasicRendering(legend);
+
+    var legendElement: D3.Selection = (<any> legend)._element;
+    var labels = legendElement.selectAll("text");
+    var swatchContainer = legendElement.select(".swatch-container");
+    var swatchContainerBCR = swatchContainer.node().getBoundingClientRect();
+
+    var lowerLabelBCR = labels[0][0].getBoundingClientRect();
+    var upperLabelBCR = labels[0][1].getBoundingClientRect();
+    assert.operator(lowerLabelBCR.right, "<=", swatchContainerBCR.left, "first label to left of swatches");
+    assert.operator(swatchContainerBCR.right, "<=", upperLabelBCR.left, "second label to right of swatches");
+
+    svg.remove();
+  });
+
+  it("renders correctly (orientation: right)", () => {
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "right");
+    legend.renderTo(svg);
+
+    assertBasicRendering(legend);
+
+    var legendElement: D3.Selection = (<any> legend)._element;
+    var labels = legendElement.selectAll("text");
+    var swatchContainer = legendElement.select(".swatch-container");
+    var swatchContainerBCR = swatchContainer.node().getBoundingClientRect();
+
+    var lowerLabelBCR = labels[0][0].getBoundingClientRect();
+    var upperLabelBCR = labels[0][1].getBoundingClientRect();
+    assert.operator(swatchContainerBCR.right, "<=", lowerLabelBCR.left, "first label to right of swatches");
+    assert.operator(swatchContainerBCR.right, "<=", upperLabelBCR.left, "second label to right of swatches");
+    assert.operator(upperLabelBCR.bottom, "<=", lowerLabelBCR.top, "lower label is drawn below upper label");
+
+    svg.remove();
+  });
+
+  it("renders correctly (orientation: left)", () => {
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "left");
+    legend.renderTo(svg);
+
+    assertBasicRendering(legend);
+
+    var legendElement: D3.Selection = (<any> legend)._element;
+    var labels = legendElement.selectAll("text");
+    var swatchContainer = legendElement.select(".swatch-container");
+    var swatchContainerBCR = swatchContainer.node().getBoundingClientRect();
+
+    var lowerLabelBCR = labels[0][0].getBoundingClientRect();
+    var upperLabelBCR = labels[0][1].getBoundingClientRect();
+    assert.operator(lowerLabelBCR.left, "<=", swatchContainerBCR.left, "first label to left of swatches");
+    assert.operator(upperLabelBCR.left, "<=", swatchContainerBCR.left, "second label to left of swatches");
+    assert.operator(upperLabelBCR.bottom, "<=", lowerLabelBCR.top, "lower label is drawn below upper label");
+
+    svg.remove();
+  });
+
+  it("re-renders when scale domain updates", () => {
+    var legend = new Plottable.Component.InterpolatedColorLegend(colorScale, "horizontal");
+    legend.renderTo(svg);
+
+    colorScale.domain([0, 85]);
+    assertBasicRendering(legend);
```

Comment:
`assertBasicRendering()` checks that the label values match the domain values. If you assume the scale updates correctly it will implicitly check this, but I agree that it'd be clearer what the test was up to if I checked the labels directly.